ADR/0011-roles-and-permissions.md

```diff
@@ -0,0 +1,81 @@
+|               | IntegrationTestScenario | appstudio.redhat.com      | get, list, watch                  | integrationtestscenarios
+|               | Enterprise contract     | appstudio.redhat.com      | get, list, watch                  | enterprisecontractpolicies
+|               | Release Strategy        | appstudio.redhat.com      | get, list, watch                  | releases, releasestrategies, releaseplans
+|               | Release Admission Plan  | appstudio.redhat.com      | get, list, watch                  | releaseplanadmissions
+|               | *JVM Build Service*     | jvmbuildservice.io        | get, list, watch                  | jbsconfigs, artifactbuilds
+|               | *Service Access*        | appstudio.redhat.com      | get, list, watch                  | spiaccesstokenbindings, spiaccesschecks, spiaccesstokens, spiaccesstokendataupdates
+|               | *Configs*               |                           | get, list, watch                  | configmaps
+|               | *Secrets*               |                           |                                   | secrets
+|               | Add User                |
+|               | User group (with SSO)   |
+| Maintainer    | Workspace               | Access to namespaces that backs workspace                           |
+|               | Application & Component | appstudio.redhat.com      | get, list, watch, create, update, patch | applications, components, componentdetectionqueries
+|               | Environment             | appstudio.redhat.com      | get, list, watch                        | promotionruns, snapshotenvironmentbindings, snapshots, environments
+|               | *GitOps*                | managed-gitops.redhat.com | get, list, watch                        | gitopsdeployments
+|               | PipelineRun             | tekton.dev                | create, get, list, watch                | pipelineruns
+|               | Pipeline Results        | results.tekton.dev        | get, list                               | results, records
+|               | IntegrationTestScenario | appstudio.redhat.com      | *                                       | integrationtestscenarios
+|               | Enterprise contract     | appstudio.redhat.com      | get, list, watch                        | enterprisecontractpolicies
+|               | Release Strategy        | appstudio.redhat.com      | *                                       | releases, releasestrategies, releaseplans
+|               | Release Admission Plan  | appstudio.redhat.com      | *                                       | releaseplanadmissions
+|               | *JVM Build Service*     | jvmbuildservice.io        | get, list, watch                        | jbsconfigs, artifactbuilds
+|               | *Service Access*        | appstudio.redhat.com      | get, list, watch                        | spiaccesstokenbindings, spiaccesschecks, spiaccesstokens, spiaccesstokendataupdates
```

Comment:
Permissions of SPI resources should be the same (not less than) Components, like:  get, list, watch, create, update, patch.
spiaccesstokendataupdates is not used by User (SA only), can be removed
spifilecontentrequest should be added

Reply:
Thanks @gazarenkov I was not sure about SPI. I will also ping Sergei to clarify a few things.